comp/api/api/utils/stream/stream.go

```diff
@@ -93,3 +106,77 @@ func GetStreamFunc(messageReceiverFunc func() MessageReceiver, streamType, agent
 		}
 	}
 }
+
+// ExportStreamLogs export output of stream-logs to a file. Currently used for remote config stream logs
+func ExportStreamLogs(la logsAgent.Component, streamLogParams *LogParams) error {
+	var f *os.File
+	var bufWriter *bufio.Writer
+
+	err := CheckDirExists(streamLogParams.FilePath)
+	if err != nil {
+		return fmt.Errorf("error creating directory for file %s: %v", streamLogParams.FilePath, err)
+	}
+
+	f, bufWriter, err = OpenFileForWriting(streamLogParams.FilePath)
+	if err != nil {
+		return fmt.Errorf("error opening file %s for writing: %v", streamLogParams.FilePath, err)
+	}
+	defer func() {
+		err := bufWriter.Flush()
+		if err != nil {
+			fmt.Printf("Error flushing buffer for log stream: %v", err)
+		}
+		f.Close()
+	}()
+
+	messageReceiver := la.GetMessageReceiver()
+
+	if !messageReceiver.SetEnabled(true) {
+		return fmt.Errorf("unable to enable message receiver, another client is already streaming logs")
+	}
+	defer messageReceiver.SetEnabled(false)
+
+	var filters diagnostic.Filters
+	done := make(chan struct{})
+	defer close(done)
+
+	logChan := messageReceiver.Filter(&filters, done)
+
+	timer := time.NewTimer(streamLogParams.Duration)
+	defer timer.Stop()
+
+	for {
+		select {
+		case log := <-logChan:
+			if _, err := bufWriter.WriteString(log + "\n"); err != nil {
+				return fmt.Errorf("failed to write to file: %v", err)
+			}
+		case <-timer.C:
+			return nil
+		}
+	}
+}
+
+// OpenFileForWriting opens a file for writing
+func OpenFileForWriting(filePath string) (*os.File, *bufio.Writer, error) {
+	log.Infof("opening file %s for writing", filePath)
+	f, err := os.OpenFile(filePath, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	if err != nil {
+		return nil, nil, fmt.Errorf("error opening file %s: %v", filePath, err)
+	}
+	bufWriter := bufio.NewWriter(f) // default 4096 bytes buffer
```

Comment:
Does the comment about the buffer size affect either parts of the code here? If it's just a piece of implementation trivia I think it's okay to remove it.